### PR TITLE
Replace .all with a function that takes language

### LINKED ISSUE #18: Use getLocalizedList?
If getLocalizedList was exported, you could for instance list out every pokemon. It would be a better solution than `require("pokemon/data/en.json")` (not even sure if that works)

I would open a pull request, but I wanted to see if this was a good idea first.

### PR DESCRIPTION
**Pull request by thesbros**
Resolves #18.

Replaced .all with a function that takes language. (or defaults to English)

This will require a major version increase as it is a breaking change.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -18,7 +17,7 @@ const languages = new Set([
 	'zh-Hant'
 ]);
 
-function getLocalizedList(lang) {
+exports.all = lang => {
```
Comment (SamVerschueren) [inline index.js]: Leave it like it was and assign `exports.all = getLocalizedList;`

Comment (thesbros) [inline index.js]: @SamVerschueren Done. :)

File: readme.md [hunk 2]
```diff
@@ -32,9 +35,18 @@ pokemon.getId('Dratini');
 
 ## API
 
-### .all: *string[]*
+### .all(lang: *string*): *string[]*
+
+Get all names.
+
+With the optional `lang` parameter you can get a localized name using a [language code](https://en.wikipedia.org/wiki/ISO_639-1):
```
Comment (SamVerschueren) [inline readme.md]: I'd suggest formatting it like this 

``````
### .all(lang: *string* = 'en'): *string[]*

Localised list of all the names.

#### lang

[Language code](#supported-languages) to retrieve the list of pokémon for.

```js
pokemon.all();
//=> ['Bulbasaur', ...]

pokemon.all('de');
//=> ['Bisasam', ...]
```
``````

But I let @sindresorhus decide what he wants.

Comment (sindresorhus) [inline readme.md]: @SamVerschueren 👍

Comment (thesbros) [inline readme.md]: @sindresorhus Latest commit look good?

File: readme.md [hunk 3]
```diff
@@ -32,9 +35,20 @@ pokemon.getId('Dratini');
 
 ## API
 
-### .all: *string[]*
+### .all(lang: *string* = 'en'): *string[]*
+
+Get a localized list of all names.
```
Comment (sindresorhus) [inline readme.md]: > Get a list of all names

The `lang` option already describes the localization.

File: test.js [hunk 4]
```diff
@@ -1,6 +1,12 @@
 import test from 'ava';
 import m from './';
 
+function testAll(t, lang, expectedNames) {
+	t.is(m.all(lang)[0], expectedNames[0]);
+	t.is(m.all(lang)[1], expectedNames[1]);
+	t.is(m.all(lang)[2], expectedNames[2]);
```
Comment (sindresorhus) [inline test.js]: This could use `t.deepEqual` and `Array#slice` for the 3 first items.

File: test.js [hunk 5]
```diff
@@ -28,6 +32,54 @@ test('.languages', t => {
...
+	'フシギダネ',
+	'フシギソウ',
+	'フシギバナ'
+]);
+
+test('Get all Russian names', testAll, 'ru', [
+	'Бульбазавр',
+	'Айвизавр',
+	'Венузавр'
+]);
```
Comment (sindresorhus) [inline test.js]: No need to test this for all localization. Testing one localization other than English is enough. That's how it's done with the other tests.

File: readme.md [hunk 6]
```diff
@@ -86,6 +104,7 @@ Pokémon names are available for the following languages:
 - `zh-Hans` (Simplified Chinese)
 - `zh-Hant` (Traditional Chinese)
 
+The language codes follow the [IETF BCP 47 standard.](https://en.wikipedia.org/wiki/IETF_language_tag)
```
Comment (sindresorhus) [inline readme.md]: This is a different link than before. Why the change?

File: readme.md (hunk 3 above)
Comment (thesbros) [inline readme.md]: Will change that.

File: test.js (hunk 5 above)
Comment (thesbros) [inline test.js]: Really? All the other tests do all languages, I was just copying what they do.

File: readme.md (hunk 6 above)
Comment (thesbros) [inline readme.md]: The previous link was to ISO 639-1, which is not actually the format used, it is BCP 47. The -Hans/-Hant is part of the IETF standard and not the ISO standard.

File: test.js (hunk 4 above)
Comment (thesbros) [inline test.js]: Something like `t.deepEqual(m.all(lang).slice(0, 4), expectedNames)`?

File: test.js (hunk 5 above)
Comment (sindresorhus) [inline test.js]: > testIDs does all languages

That must be an oversight on my part. I'll remove those after this PR.

> I can change it to test just English & German though.

👍

File: test.js (hunk 4 above)
Comment (sindresorhus) [inline test.js]: Yup

Comment (SamVerschueren) [inline test.js]: Yes

**Comment by thesbros** [comment]
All those changes done. :)

File: readme.md [hunk 7]
```diff
@@ -44,7 +58,9 @@ Random name.
 
 Get name from ID.
 
-With the optional `lang` parameter you can get a localized name using a [language code](https://en.wikipedia.org/wiki/ISO_639-1):
+#### lang
+
+[Language code](#supported-languages) to retrieve the list of Pokémon for.
```
Comment (SamVerschueren) [inline readme.md]: > [Language code](#supported-languages) to retrieve the Pokémon for.

File: readme.md [hunk 8]
```diff
@@ -57,7 +73,9 @@ pokemon.getName(100, 'de');
 
 Get ID from name.
 
-With the optional `lang` parameter you can get the ID for a localized name using a [language code](https://en.wikipedia.org/wiki/ISO_639-1):
+#### lang
+
+[Language code](#supported-languages) to retrieve the list of Pokémon for.
```
Comment (SamVerschueren) [inline readme.md]: > [Language code](#supported-languages) of the localized name .

Comment (thesbros) [inline readme.md]: Woops! Fixed.

**Comment by sindresorhus** [comment]
Thanks :)
